Add  enhancements US31 #20 and US33 #21

piglet25 opened the pull request:
Here is the code for sprint #3 issue number #20 and issue number #21

JackieDreamy commented:
Merge it, but some of stuffs need to be refactored, let us discuss after the class.

piglet25 commented:
Hi Yanfei,

I did the best I could.

First in order to know if someone has been married they will be in two 
families. One family when they are born and the a second family when 
they are married. So if not married only in one family. No way to check 
this in current structure that I could find. So I added all child ids to 
a list and deleted them if they showed up as a mother or father later.

Then I found I could not use the normal parseChild because I needed 
extra information. For the not married I would need to pass the 
unmarried list because some children may be married but not others. In 
the orphan case I need to know how old they were when orphaned which is 
the later of mother's and father's death dates. There was also no way to 
pass that to parseChild as some children may be 18+ when last parent 
died. So I just checked this in place. I didn't want to change any of 
your structure. I knew you would prefer to do make those kind of 
changes.

Thanks,
Susan

On 10/28/2015 11:01 AM, JackieDreamy wrote:
> Merge it, but some of stuffs need to be refactored, let us discuss
> after the class.
> 
> --
> Reply to this email directly or view it on GitHub [1].
> 
> 	*
> 
> Links:
> ------
> [1] 
> https://github.com/JackieDreamy/AgileSprint/pull/54#issuecomment-151873286

JackieDreamy commented:
I will tell you why I said your code need to be refactored after the class.:)
